Core/App.Web/Middleware/ErrorHandlingMiddleware.cs

```diff
@@ -51,32 +62,60 @@ public async Task Invoke(HttpContext context)
         private Task HandleExceptionAsync(HttpContext context, Exception exception)
         {
             var exceptionMessage = GetExceptionMessage(exception);
+            var exceptionType = exception.GetType();
+            var serializedData = JsonSerializer.Serialize(exception.Data);
+            var correlationId = correlationContext.CorrelationContext.CorrelationId;
+            HttpStatusCode exceptionCode = HttpStatusCode.InternalServerError;
             
-            var basicExceptionResult = JsonSerializer.Serialize(new 
-            { 
-                message = exceptionMessage, data = exception.Data 
-            });
-            var fullExceptionResult = JsonSerializer.Serialize(new 
-            { 
-                message = exceptionMessage, data = exception.Data, exception = exception.ToString() 
-            });
 
-            if (_exceptionTypes.TryGetValue(exception.GetType().Name, out HttpStatusCode exceptionCode))
+            foreach (var kv in _exceptionTypes)
+            {
+                if (exceptionType.Equals(kv.Key) || exceptionType.IsSubclassOf(kv.Key))
+                {
+                    exceptionCode = kv.Value;
+                    break;
+                }
+            }
+
+            if (exceptionCode != HttpStatusCode.InternalServerError)
             {
                 // log as warning the expected exceptions
-                _logger.LogWarning($"Expected exception ({(int)exceptionCode})-{exceptionCode}: {basicExceptionResult}"); 
+                _logger.LogWarning(exception, $"Exception: {exceptionType.Name} (expected): HTTP Status: {(int)exceptionCode} ({exceptionCode}): {exceptionMessage} (data: {serializedData})"); 
             }
             else
             {
                 // log as error the unexpected exceptions
                 exceptionCode = HttpStatusCode.InternalServerError; 
-                _logger.LogError(exception, $"Unexpected exception ({(int)exceptionCode})-{exceptionCode}");
+                _logger.LogError(exception, $"Exception: {exceptionType.Name} (unexpected): HTTP Status: {(int)exceptionCode} ({exceptionCode}): {exceptionMessage} (data: {serializedData})");
             }
 
-            string result = _webHostEnvironment.IsDevelopment() ? fullExceptionResult : basicExceptionResult;
             
             context.Response.ContentType = "application/json";
             context.Response.StatusCode = (int)exceptionCode;
+            string result = string.Empty;
+
+            if (_webHostEnvironment.IsDevelopment())
+            {
+                result = JsonSerializer.Serialize(new ErrorResponse
+                {
+                    IsError = true,
```

Comment:
why do you need the IsError property since is always set as true? Also, this is an error response, so it always contains an error.

Reply:
Make it somewhat future compatible with a general Result object

Reply:
So, ErrorResponse could then derive from an 'BaseResponse' that would hold that IsError property. By default that would be false, but overriding classes could set it to true, or a user may set it to true if they want the result-object to be treated as an error. What use-cases are you thinking of here @meghuizen?

Reply:
That was the idea. However I just removed the IsError attribute all together. Since we don't return any 200 result at all when returning an error.